components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -3277,9 +3282,13 @@ private static String getThumbPrintWithAlgorithm(Certificate certificate, String
             byte[] der = certificate.getEncoded();
             digestValue.update(der);
             byte[] digestInBytes = digestValue.digest();
-            String publicCertThumbprint = hexify(digestInBytes);
+            // According to the RFC7515#section-4.1.8 converting bytes into hex is not required.
+            // Added the config for the backward compatibility.
+            if (Boolean.parseBoolean(IdentityUtil.getProperty(JWKS_IS_HEXIFY_REQUIRED))) {
```

Comment:
This util seems to be used in other places as well(public util). The fix seems to be for an issue specific to jwks representations. We need to make sure this config doesn't impact other places where this util is being used. 
This place might not be the optimal place to have this check.

Reply:
Added the config to the JWKSEndpoint specific method
Fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2331/commits/83d7809e14bd6a2f711edb9c9bd516cae60aaac7